### DISCUSSION TITLE
Cannot read properties of undefined (reading 'call')

### DISCUSSION THREAD

**Discussion by Musadiqkhan-nullbrainer** [webpack/webpack#15369] [Q&A]
# Bug report

**What is the current behavior?**
Upgraded to latest webpack version from webpack 5.23.0 and encountered a runtime error `Cannot read properties of undefined (reading 'call') at __webpack_require__ (app.8dfc72f16583797bed7d.js:2:1594857)`. Downgraded to find out the issue starts showing up from webpack 5.24.1

This is most probably due to **_improve performance of finding modules to concatenate_**. If I set optimization.concatenateModules to false then everything works fine. 

I have found out list of outdated packages using `npm outdated`. Updated them to wanted version and still facing the issue.

How can I find the real cause?

**Other relevant information:**
webpack version:  5.68.0
Node.js version: 
Operating System: Windows
Additional tools:

**Comment by alexander-akait** [comment]
Please create reproducible test repo

**Comment by Musadiqkhan-nullbrainer** [comment]
Not possible because it's a client's project and I know for a fact issue is related to this specific project because for another project I am running the latest webpack without issues.

@alexander-akait I need to know way to debug it. the error it gives is not clear. Is there anyway I can get a proper error which I can understand and fix?

**Reply by vankop** [reply]
there is concatenated module in webpack https://github.com/webpack/webpack/blob/main/lib/optimize/ConcatenatedModule.js `codeGeneration` method generates code.. but without reducing your issue to reproducible repo hard to guess..

**Reply by Musadiqkhan-nullbrainer** [reply]
@vankop Is it a good approach to use optimization.concatenateModules: false?

**Reply by vankop** [reply]
this will increase your output size. if it is ok for you, you can use it.

**Comment by bmarti44** [comment]
@Musadiqkhan-nullbrainer we had this same exact issue (not just the error, but also downgrading to 5.24.0 versus 5.24.1 would fix things). in our case we had the `ModuleConcatenationPlugin` defined twice (when mode is production and in the plugins block) which caused the issue. More info https://github.com/webpack/webpack/issues/17100 

you may have some plugin causing an issue in your production config, try disabling them one by one and see if the error persists.

**Reply by EvanSanderson** [reply]
Had the same issue after a recent upgrade to Webpack 5. To reiterate @bmarti44's point (or, at least, what I think might be the issue):

If you have the ModuleConcantenationPlugin included in your webpack config's plugin section like this: 

`new webpack.optimize.ModuleConcatenationPlugin()`

and that same config has mode set to `production`, then I believe this issue might occur, since according to the documentation: "By default this plugin is already enabled in [production mode](https://webpack.js.org/configuration/mode/#mode-production) and disabled otherwise." It does seem a little odd that this would be the case, but removing the plugin instantiation fixed things. 

Would it be possible to add this detail to the documentation (either [here](https://webpack.js.org/plugins/module-concatenation-plugin/) or in the Webpack 5 upgrade docs)? I'd be happy to do so if it's easier

**Reply by alexander-akait** [reply]
@EvanSanderson Feel free to do it
